core/services/workflows/engine.go

```diff
@@ -770,10 +777,19 @@ func (e *Engine) workerForStepRequest(ctx context.Context, msg stepRequest) {
 		Ref:         msg.stepRef,
 	}
 
-	// TODO ks-462 inputs
 	logCustMsg(ctx, cma, "executing step", l)
 
+	stepExecutionStartTime := time.Now()
 	inputs, outputs, err := e.executeStep(ctx, l, msg)
+	stepExecutionDuration := time.Since(stepExecutionStartTime).Seconds()
+
+	curStep, verr := e.workflow.Vertex(msg.stepRef)
+	if verr != nil {
+		l.Errorf("failed to resolve step in workflow; error %v", err)
+		return
```

Comment:
It's really important not to return here: you need to fetch this for a monitoring concern; failing to fetch it shouldn't impede the workflow as a whole from running.

Reply:
305b1f32c74c4f19bfdb80f657a76f1ed977abaa